### ISSUE TITLE
Remove `x^y` and use `x.ipow(y)` instead?

### ISSUE THREAD

**Issue by konnov** [informalsystems/quint#579]
As we saw in #578, `x^y` may have a somewhat unexpected priority. Moreover, C-like languages often define `x^y` to be a bit operator. Given all these complications and potential surprises, I would prefer to remove `x^y` in favor of `x.ipow(y)`, which we already have. For instance, Rust has the method [pow](https://doc.rust-lang.org/std/primitive.i8.html#method.pow), which could be used as `x.pow(y)`.

/cc @bugarela, @shonfeder, @thpani, @Kukovec, thoughts?

**Comment by konnov** [comment]
I just discovered that `x ** y` has *lower* priority than unary `-` in Solidity, according to their [cheatsheet](https://docs.soliditylang.org/en/v0.8.17/cheatsheet.html). We should definitely remove `x^y` to avoid confusion.

**Comment by thpani** [comment]
Sounds good.

Just to clarify, does the `i` in `ipow` stand for integer?
The only language I've found calling it `ipow` is ABAP 😏

**Comment by shonfeder** [comment]
Seems fine to switch to me. Just using `pow` seems simplest?

**Comment by konnov** [comment]
true. We have normal-form names for integer operators that start with `i`, that is, `iadd`, `isub`. There are two reasons:

 - Avoid obvious name clashes, as `add` and `sub` are good names for user-defined operators.
 - Once we fix the module system and introduce file imports, I expect new nice modules to appear quickly, for instance, the modules for using fixed bitwidth integers. Then, we will have operators like `i32add`, `i64add`, etc.

I agree that `pow` would be the best solution, but than it will be incompatible with the naming scheme for the other arithmetic operators.

Actually, we could expose `+`, `-`, etc. by default and keep the standard normalized names fully qualified like `arith::add` and `arith::sub`. That will probably solve the problem of obvious names clashes.

**Comment by thpani** [comment]
> Once we fix the module system and introduce file imports, I expect new nice modules to appear quickly, for instance, the modules for using fixed bitwidth integers. Then, we will have operators like `i32add`, `i64add`, etc.

As you suggested, I'd prefer proper namespacing over Hungarian notation.

But given current `iadd` and `isub`, I'm okay with `ipow` until we have that namespacing capability.
